From 453257f68148ca79e62886cf23c2be2a84b694f9 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Wed, 8 Nov 2017 22:50:50 +0000
Subject: [PATCH] cmd/snap-confine: Respect biarch nature of libdirs

This follows the same bi-arching seen in other parts of the policy and
ensures snap-update-ns rules are actually matched on biarch distributions
employing a lib64 tree.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 cmd/snap-confine/snap-confine.apparmor.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/cmd/snap-confine/snap-confine.apparmor.in b/cmd/snap-confine/snap-confine.apparmor.in
index 2e1681b7a..0276b8ef7 100644
--- a/cmd/snap-confine/snap-confine.apparmor.in
+++ b/cmd/snap-confine/snap-confine.apparmor.in
@@ -440,8 +440,8 @@
     profile snap_update_ns (attach_disconnected) {
         # The next four rules mirror those above. We want to be able to read
         # and map snap-update-ns into memory but it may come from a variety of places.
-        /usr/lib{,exec}/snapd/snap-update-ns mr,
-        /var/lib/snapd/hostfs/usr/lib{,exec}/snapd/snap-update-ns mr,
+        /usr/lib{,exec,64}/snapd/snap-update-ns mr,
+        /var/lib/snapd/hostfs/usr/lib{,exec,64}/snapd/snap-update-ns mr,
         /{,var/lib/snapd/}snap/core/*/usr/lib/snapd/snap-update-ns mr,
         /var/lib/snapd/hostfs/{,var/lib/snapd/}snap/core/*/usr/lib/snapd/snap-update-ns mr,
 
-- 
2.15.0

